modules/apps/portal/portal-context/src/main/java/com/liferay/portal/context.internal/CompanyInvocationContextProviderImpl.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.context.internal;
+
+import com.liferay.portal.kernel.context.CompanyInvocationContextProvider;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.security.auth.CompanyThreadLocal;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.util.CompanyThreadLocal;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Jorge Ferrer
+ */
+@Component(service = CompanyInvocationContextProvider.class)
+public class CompanyInvocationContextProviderImpl
+	implements CompanyInvocationContextProvider {
+
+	@Override
+	public Company getCurrent() {
+		try {
+
+			// TODO: What is the best way to handle when the current invocation
```

Comment:
I would use fetchComany() and return null.

Reply:
Same with Group.

Reply:
Makes sense. 

Thanks for the suggestion.

Reply:
I've added a commit introducing a method called `isPresent()` to allow checking if an object is available in the current invocation context.